crates/ethereum/src/lib.rs

```diff
@@ -182,34 +212,97 @@ impl EthereumApi for EthereumClient {
                         self.pending_state_updates.remove(&eth_block);
                     }
                 }
-                Some(log) = logs.next() => {
-                    // Decode the message
-                    let log: Log<StarknetCoreContract::LogMessageToL2> = log.log_decode()?;
-                    // Create L1ToL2MessageHash from the log data
-                    let msg = L1ToL2MessageLog {
-                        message_hash: H256::from(log.inner.message_hash().to_be_bytes()),
-                        l1_tx_hash: log.transaction_hash.map(|hash| H256::from(hash.0)).unwrap_or_default(),
-                    };
-                    // Emit the message log
-                    callback(EthereumEvent::MessageLog(msg)).await;
-                }
                 Some(block_number) = finalized_block_rx.recv() => {
                     // Collect all state updates up to (and including) the finalized block
                     let pending_state_updates: Vec<EthereumStateUpdate> = self.pending_state_updates
-                        .range(..=block_number.get())
+                        .range(..=block_number)
                         .map(|(_, &update)| update)
                         .collect();
                     // Remove emitted updates from the map
-                    self.pending_state_updates.retain(|&k, _| k > block_number.get());
+                    self.pending_state_updates.retain(|&k, _| k > block_number);
                     // Emit the state updates
                     for state_update in pending_state_updates {
-                        callback(EthereumEvent::StateUpdate(state_update)).await;
+                        let _ = callback(state_update).await;
                     }
                 }
             }
         }
     }
 
+    async fn get_l1_handler_txs(
+        &self,
+        chain: EthereumChain,
```

Comment:
Instead of the chain the core contract address should be an argument here. We already have that in `L1SyncContext` anyway and the value of that address is correct for custom chains as well.